src/main/java/duck/Ui.java

```diff
@@ -0,0 +1,128 @@
+package duck;
+import java.util.Scanner;
+
+/**
+ * Handles user interactions, including reading input and displaying messages.
+ */
+public class Ui {
+
+    private final Scanner scanner;
+
+    /**
+     * Constructs a Ui instance and initializes the Scanner for user input.
+     */
+    public Ui() {
+        scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Reads the next command entered by the user.
+     *
+     * @return The user input as a String.
+     */
+    public String readCommand() {
+        return scanner.nextLine();
+    }
+
+    /**
+     * Displays the welcome message to the user.
+     */
+    public String showWelcome() {
+        return """
+            Hello! I'm DUCK
+            What can I do for you?");
+            Enter command 'list' to see all saved tasks (if any)""";
+    }
+
+    /**
+     * Displays the list of tasks.
+     *
+     * @param list The task list to be displayed.
+     */
+    public String showList(TaskList list) {
+        StringBuilder output = new StringBuilder();
+        output.append("Here are the tasks in your list:\n");
+        for (int i = 0; i < list.size(); i++) {
+            output.append("  ").append(i + 1).append(". ").append(list.get(i).toString()).append("\n");
+        }
+        return output.toString();
+    }
+
+    /**
+     * Displays a message indicating a task has been marked as completed.
+     *
+     * @param t The task that was marked as completed.
+     */
+    public String mark(Task t) {
+        return "Nice! I've marked this task as done:\n"
+               + "     [" + t.getStatusIcon() + "] " + t.getDescription() + "\n";
```

Comment:
nit: Should line wrapping start with 8 spaces? I think alignment with previous line is fine too.

```suggestion
                + "     [" + t.getStatusIcon() + "] " + t.getDescription() + "\n";
```